app/components/Preview/Types/PreviewUri.tsx

```diff
@@ -12,11 +13,14 @@ export type PreviewUriProps = {
 };
 
 export function PreviewUri(props: PreviewUriProps) {
-  const previewFetcher = useFetcher<PreviewResult>();
+  const { isOnline, onceOnline } = useNetworkState();
+  const previewFetcher = useFetcher < PreviewResult > ();
 
   useEffect(() => {
```

Comment:
I actually think we don't even need the useEffect call here, and all the code inside the useEffect callback can be hoisted just to the top level of the component function